python/ray/data/datasource/lance_datasource.py

```diff
@@ -36,32 +36,32 @@ def __init__(
         self.storage_options = storage_options
 
         self.lance_ds = lance.dataset(uri=uri, storage_options=storage_options)
-        self.fragments = self.lance_ds.get_fragments()
+        fragments = self.lance_ds.get_fragments()
+        self.fragment_ids = [f.metadata.id for f in fragments]
+        self.schema = fragments[0].schema if len(fragments) > 0 else None
 
     def get_read_tasks(self, parallelism: int) -> List[ReadTask]:
         read_tasks = []
-        for fragments in np.array_split(self.fragments, parallelism):
-            if len(fragments) <= 0:
+        for fragment_ids in np.array_split(self.fragment_ids, parallelism):
+            if len(fragment_ids) <= 0:
                 continue
 
-            num_rows = sum([f.count_rows() for f in fragments])
-            input_files = [
-                data_file.path() for f in fragments for data_file in f.data_files()
-            ]
-
```

Comment:
I think you can still keep these. We've established that `count_rows` is not the slow part. In fact, it's even faster than `get_fragments()`.

Also, small nit: you don't need the `[]` inside of sum. If you omit them you get a generator expression which bypasses the need to allocate the whole list.

```python
num_rows = sum(f.count_rows() for f in fragments)
```